### PR TITLE
SyntaxHighlighter: Put formatted code to clipboard

### PR DESCRIPTION
**Pull request by mbelsky**
## What I did

I've added some changes to put formatted code to clipboard if `SyntaxHighlighter.format=true`

## How to test

1. Open ?path=/story/basics-syntaxhighlighter--bordered-copy-able in examples/official-storybook
1. Click 'Copy' button
1. Check that your clipboard equals to
```
import { Good, Things } from 'life';

const result = () => <Good><Things /></Good>;

export { result as default };
```

### CODE REVIEW

File: lib/components/src/syntaxhighlighter/syntaxhighlighter.tsx
Comment (ndelangen) [inline lib/components/src/syntaxhighlighter/syntaxhighlighter.tsx:127]: trim() should not happen before formatting

**Review by ndelangen — CHANGES_REQUESTED** [review]

File: lib/components/src/syntaxhighlighter/syntaxhighlighter.tsx
Comment (mbelsky) [reply lib/components/src/syntaxhighlighter/syntaxhighlighter.tsx:127]: The previous implementation do the following steps:
1. cast children to string
2. trim
3. formatting

https://github.com/storybookjs/storybook/pull/11276/files#diff-315d7b1ff36cd2df0bfa65e00d60dd9cL153

Does it mean the previous implementation had same issue?

Comment (ndelangen) [reply lib/components/src/syntaxhighlighter/syntaxhighlighter.tsx:127]: I thought the previous implementation called trim on the result AFTER formatting?

Comment (mbelsky) [reply lib/components/src/syntaxhighlighter/syntaxhighlighter.tsx:127]: The previous implementation had so many parenthesis that might be misleading. `formatter((children as string).trim())` same as `formatter(children.trim())`

Comment (mbelsky) [reply lib/components/src/syntaxhighlighter/syntaxhighlighter.tsx]: As I've found in formatter's `tests` (https://github.com/storybookjs/storybook/blob/da777b8e4bed32ce5ec0b1b60f661e913d8d2f39/lib/components/src/syntaxhighlighter/formatter.test.js#L29-L38) there is no reason to trim a string before or after formatting.

File: lib/components/src/syntaxhighlighter/syntaxhighlighter.tsx
Comment (mbelsky) [inline lib/components/src/syntaxhighlighter/syntaxhighlighter.tsx:127]: I've added this check `!children.trim()` to also return `null` instead of a component for strings which contain whitespaces only.

File: lib/components/src/syntaxhighlighter/syntaxhighlighter.tsx
Comment (mbelsky) [reply lib/components/src/syntaxhighlighter/syntaxhighlighter.tsx]: Based on formatter's tests I think there is no reason to call trim in `SyntaxHighlighter` component. @ndelangen what are your thoughts?

**Review by ndelangen — APPROVED** [review]

File: lib/components/src/syntaxhighlighter/syntaxhighlighter.tsx
Comment (ndelangen) [reply lib/components/src/syntaxhighlighter/syntaxhighlighter.tsx]: 👍

**Comment by ndelangen** [comment]
@tmeasday could you check this change?

**Review by tmeasday — APPROVED** [review]
LGTM

**Comment by shilman** [comment]
Since this is a 6.1 feature, let's not merge this until 6.0 has stabilized. I'll merge once it's OK.

**Pull request merged**
